src/main/java/nextstep/subway/line/domain/Sections.java

```diff
@@ -20,128 +20,225 @@ public Sections(List<Section> sections) {
         sections = sections;
     }
 
-    /* 도메인 기능 메서드 */
-
-    // 노선의 역 조회
     public List<Station> getStations() {
         List<Station> stations = new ArrayList<>();
-        if(getSectionSize() > 0){
-            Station downStation = getFirstStation();
-            stations.add(downStation);
-
-            while (downStation != null) {
-                Station finalDownStation = downStation;
-                Optional<Section> nextLineStation = sections.stream()
-                        .filter(it -> it.getUpStation() == finalDownStation)
-                        .findFirst();
-                if (!nextLineStation.isPresent()) {
-                    break;
-                }
-                downStation = nextLineStation.get().getDownStation();
-                stations.add(downStation);
-            }
+        if(getSectionSize() == 0) {
+            return stations;
+        }
+
+        Station station = getFirstStation();
+        stations.add(station);
+        while ((station = getNextStation(station)) != null){
+            stations.add(station);
         }
+
         return stations;
     }
 
-    // 구간개수
+    public Station getNextStation(Station station) {
+        Station nextStation = null;
+        if (sections.size() == 0) {
+            return nextStation;
```

Comment:
```suggestion
            return null;
```

nextStation이 NULL인지 한번더 확인할 필요없이 바로 null이라고 명시적으로 적어주면 더 좋을 것 같아요